File: src/test/java/com/github/liebharc/queryenrichment/InMemoryPlanBuilder.java
package com.github.liebharc.queryenrichment;

import java.util.List;

public class InMemoryPlanBuilder extends PlanBuilder {

    public InMemoryPlanBuilder(List<Step<?>> steps) {
        super(steps);
    }

    @Override
    protected QueryBuilder getQueryBuilder() {
        return new InMemoryQueryBuilder();
    }

    @Override
    protected boolean isSupportedByQuery(SimpleExpression criteria) {
        return false;
    }
}

File: src/test/java/com/github/liebharc/queryenrichment/InMemoryPlanBuilder.kt
package com.github.liebharc.queryenrichment

class InMemoryPlanBuilder(steps: List<ExecutableStep<*, Any?>>) : PlanBuilder<Any?>(steps) {

    override val queryBuilder: QueryBuilder
        get() = InMemoryQueryBuilder()

    override fun isSupportedByQuery(criteria: SimpleExpression<*>): Boolean {
        return false
    }
}
